inttest/basic/basic_test.go

```diff
@@ -111,6 +114,19 @@ func (s *BasicSuite) TestK0sGetsUp() {
 		s.Require().NoError(common.VerifyKubeletMetrics(ctx, kc, node))
 	}
 
+	s.T().Log("checking kube-router gobgp functionality")
+	kubeRouterPods, err := kc.CoreV1().Pods("kube-system").List(ctx, metav1.ListOptions{LabelSelector: "k8s-app=kube-router"})
+	s.Require().NoError(err)
+	// Just take the first running pod for execing the gobgp command
+	for _, pod := range kubeRouterPods.Items {
+		if pod.Status.Phase == corev1.PodRunning {
```

Comment:
Don't we need to ensure that we have at least one running pod? Otherwise this test would succeed without checking anything.

Reply:
The test verifies all kube-router pods are running already before this

Reply:
Saw that. Just thinking that it might have died in the meantime. Anyhow, very unlikely.